### PR TITLE
Patch necessary to make FP8 convolution compile with hiprtc

### PR DESCRIPTION
**Pull request by umangyadav**
With this set of minimal changes, compilation succeeds without running into `__ptrdiff_t` problem when using hiprtc. 

Testing:
Ran  fp8e4m3fnuz (F8 with FNUZ) convoltion verification tests inside MIGraphX that calls into MIOpen for convolution on MI300., with mainline builds.
 
It passes verification tests. Two of the verification tests are flaky though. They sometime pass, sometime don't.  Those tests always pass with rocMLIR so there could potentially be a bug in MIOpen. Identifying problem and putting fix for it is not motive of this PR though.  This PR is mainly to fix compilation problem. 

Running `make check` on MIOpen fails some tests with same `__ptrdiff_t` problem though because some other testing files are still including `limits`.  Failed tests are same as failed tests on `develop` branch on MI300. 

for example : https://github.com/ROCmSoftwarePlatform/MIOpen/blob/762ae81872fa06f925860bce7158fa912e0b0e3b/src/kernels/MIOpenCheckNumerics.cpp#L73 

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/762ae81872fa06f925860bce7158fa912e0b0e3b/src/composable_kernel/composable_kernel/include/utility/data_type.hpp#L14

```
	The following tests FAILED:
	          9 - test_check_numerics_test (Failed)
	         42 - test_reduce_test (Failed)
	        796 - ConvBwdTest/ConvBwdSolverTestF8.CKConvF8Bwd/(5,  G:1 N:16 C:16 D:1 H:14 W:14 k:16 z:1 y:3 x:3 pad_z:0 pad_y:1 pad_x:1 stride_z:1 stride_y:1 stride_x:1 dilation_z:1 dilation_y:1 dilation_x:1 conv_mode:0, 8) (Failed)
	        808 - DUMP_TENSOR_TEST.testDump_NAN_float (Failed)
	        810 - DUMP_TENSOR_TEST.testDump_NAN_half_float (Failed)
        812 - DUMP_TENSOR_TEST.testDump_NAN_bfloat16 (Failed)
```

Fixes for those should come later as some other PR.

### CODE REVIEW

**Review by JehandadKhan — APPROVED** [review]
LGTM

**Comment by junliume** [comment]
@umangyadav So this fixes compilation issues with updated hipRTC; however, several tests which you mentioned are still failing?

**Comment by umangyadav** [comment]
> @umangyadav So this fixes compilation issues with updated hipRTC; however, several tests which you mentioned are still failing?

Those tests fail on develop branch as well.  Fixing those would require bigger overhaul in multiple files. 

For now, i've only focused on bare minimum to have FP8 compile with hipRTC.

**Comment by junliume** [comment]
> > @umangyadav So this fixes compilation issues with updated hipRTC; however, several tests which you mentioned are still failing?
> 
> Those tests fail on develop branch as well. Fixing those would require bigger overhaul in multiple files.
> 
> For now, i've only focused on bare minimum to have FP8 compile with hipRTC.

This PR may need to be combined with https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2519 with this one going in first. Also it might be a large amount of work if we need to patch <limits> everywhere.

I am testing via the latest mainline docker. 

Previously, we did not change F8 header voluntarily expecting that there will be a uniformed one very soon.

BTW~ these breaking changes from hipRTC, how can they be merged in mainline in the first place?

**Comment by junliume** [comment]
@atamazov need your blessing on this hipRTC workaround too.

**Review by junliume — APPROVED** [review]

File: src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp
Comment (atamazov) [inline src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp:73]: [small problem] Actually this should be enclosed into 
```C++
#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
#include <limits>
#endif
```
See #627. Otherwise HIP builds via COMGR (i.e. without HIPRTC) would still encounter the same problem. I'll fix this.

File: src/kernels/hip_float8.hpp
Comment (atamazov) [inline src/kernels/hip_float8.hpp:554]: This should be:
```C++
return !(x.is_nan() || x.is_inf());
```
isn't it?

File: src/kernels/hip_float8.hpp
Comment (atamazov) [inline src/kernels/hip_float8.hpp:558]: ditto

**Review by atamazov — COMMENTED** [review]
@umangyadav @JehandadKhan @junliume @pfultz2 FYI

File: src/kernels/hip_float8.hpp
Comment (atamazov) [inline src/kernels/hip_float8.hpp:549]: It should be 
```C++
#ifdef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
```

**Comment by atamazov** [comment]
@junliume 
> need your blessing on this hipRTC workaround too.

Just to be clear, this is not a workaround. This is a kind of adaptation for HIPRTC/COMGR limitations. Otherwise I would definitely request for appropriate formalization ;)

I recommend removing https://github.com/ROCm/MIOpen/labels/workaround.

File: src/kernels/hip_float8.hpp
Comment (umangyadav) [reply src/kernels/hip_float8.hpp:554]: Yes i copy pasted previous logic without realizing this.  This should be fixed

**Comment by atamazov** [comment]
@umangyadav 
> ```
> The following tests FAILED:...
> ```
> Fixes for those should come later as some other PR.

Am I right that the failures got out of the way before this PR is merged in?

**Comment by umangyadav** [comment]
> Am I right that the failures got out of the way before this PR is merged in? 

I still saw them before PR was merged on MI300 as far as i recall

**Pull request merged**
